### PR TITLE
[macOS export] Simplify code signing options, add support for rcodesign tool for signing and notarization.

### PR DESCRIPTION
**Pull request by bruvzg**
- Automatically set `timestamp` and `hardened runtime` options depending on signature type.
- Adds support for `rcodesign` signing and notarization.
- Adds support for hiding irrelevant export options depending on selected codesign and notarization tools.
- Adds `password` property hint.

Screenshot 2022-08-10 at 12 44 43

Implements https://github.com/godotengine/godot-proposals/issues/5115

### CODE REVIEW

**Comment by Calinou** [comment]
For the password hint, you can probably use `String::utf8("•")` for the individual characters instead of `*`. This will make it look more like password fields in modern OSes :slightly_smiling_face:

**Comment by bruvzg** [comment]
> For the password hint, you can probably use String::utf8("•") for the individual characters instead of *. The editor font should have this character already (it's used for bullet point lists in node configuration warnings). This will make it look more like password fields in modern OSes 🙂

In 3.x, we have dynamic font with this character as a default theme font, so I guess it a good to change to the default value for the `LineEdit` in general - https://github.com/godotengine/godot/pull/64267

File: platform/macos/export/export_plugin.cpp [hunk 1]
```diff
@@ -51,11 +51,51 @@ void EditorExportPlatformMacOS::get_preset_features(const Ref<EditorExportPreset
...
+			default: { // disabled
+				if (p_option == "codesign/identity" || p_option == "codesign/certificate_file" || p_option == "codesign/certificate_password" || p_option == "codesign/custom_options" || p_option.begins_with("codesign/entitlements")) {
+					return false;
+				}
+			} break;
+		}
+
+		// Hide irrelevant notarization options.
+		int notray_tool = p_preset->get("notarization/notarization");
+		switch (notray_tool) {
```
Comment (akien-mga) [inline platform/macos/export/export_plugin.cpp:85]:
```suggestion
		int notary_tool = p_preset->get("notarization/notarization");
		switch (notary_tool) {
```

File: platform/macos/export/export_plugin.cpp [hunk 2]
```diff
@@ -415,156 +466,284 @@ void EditorExportPlatformMacOS::_fix_plist(const Ref<EditorExportPreset> &p_pres
  */
 
 Error EditorExportPlatformMacOS::_notarize(const Ref<EditorExportPreset> &p_preset, const String &p_path) {
+	int notray_tool = p_preset->get("notarization/notarization");
+	switch (notray_tool) {
```
Comment (akien-mga) [inline platform/macos/export/export_plugin.cpp:470]:
```suggestion
	int notary_tool = p_preset->get("notarization/notarization");
	switch (notary_tool) {
```

File: platform/macos/export/export_plugin.cpp [hunk 3]
```diff
@@ -1598,65 +1790,98 @@ bool EditorExportPlatformMacOS::has_valid_project_configuration(const Ref<Editor
...
-	if (!ad_hoc && !FileAccess::exists("/usr/bin/codesign") && !FileAccess::exists("/bin/codesign")) {
-		err += TTR("Warning: Xcode command line tools are not installed, using built-in \"codesign\". Code signing is limited to ad-hoc signature only.") + "\n";
+		case 3: { // "codesign"
+			ad_hoc = (p_preset->get("codesign/identity") == "" || p_preset->get("codesign/identity") == "-");
+		} break;
+#endif
+		default: {
+		};
 	}
+	int notray_tool = p_preset->get("notarization/notarization");
```
Comment (akien-mga) [inline platform/macos/export/export_plugin.cpp:1810]: Same typo.

**Review by akien-mga — APPROVED** [review]
Looks amazing!

If rcodesign works well for signing and notarization, we could consider in the future making it the default and adding logic to download and configure it on demand. So it would mostly work out of the box for users. (The same logic could be applied for rcedit too.)

Needs a rebase.

**Comment by akien-mga** [comment]
Thanks!

File: editor/editor_property_name_processor.cpp [hunk 4]
```diff
@@ -237,6 +242,7 @@ EditorPropertyNameProcessor::EditorPropertyNameProcessor() {
 	capitalize_string_remaps["webp"] = "WebP";
 	capitalize_string_remaps["webrtc"] = "WebRTC";
 	capitalize_string_remaps["websocket"] = "WebSocket";
+	capitalize_string_remaps["wine"] = "wine";
```
Comment (aaronfranke) [inline editor/editor_property_name_processor.cpp:245]: The official way to capitalize [Wine](https://www.winehq.org/) is "Wine", why was this changed?
